### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by rotsen18**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,34 +1,108 @@
...
+                self.decks.append(Deck(start[0], column))
+        else:
+            for row in (range(start[0], end[0] + 1)):
+                self.decks.append(Deck(start[1], row))
+
+    def _create_all_decks(self, coordinates):
+        self.decks = [
+            Deck(row_index, col_index)
+            for row_index, col_index in coordinates
+        ]
```
Comment (Magus6784) [inline app/main.py:26]: You never used it

File: app/main.py [hunk 2]
```diff
@@ -1,34 +1,108 @@
...
+                for col_delta in range(-1, 2):
+                    check_coords = (row + row_delta, col + col_delta)
+                    ship_coords = [
+                        (deck.row, deck.col)
+                        for deck in ship.decks
+                    ]
+                    if (check_coords in self.field
+                            and check_coords not in ship_coords):
+                        raise Exception("Ships shouldn't be located"
+                                        "in the neighboring cells")
```
Comment (Magus6784) [inline app/main.py:93]: Better create specific error and don't use Exception

**Review by Magus6784 — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (rotsen18) [reply app/main.py:26]: Missed it after refactoring

File: app/main.py (hunk 2 above)
Comment (rotsen18) [reply app/main.py:93]: If I were writing a full implementation of the game, I would create a new BattleError parent class with all kinds of exceptions. But here I decided enough basic exception and description to it.

**Review by Magus6784 — APPROVED** [review]

File: app/main.py [hunk 3]
```diff
@@ -19,12 +19,6 @@ def _create_ship(self, start, end):
             for row in (range(start[0], end[0] + 1)):
                 self.decks.append(Deck(start[1], row))
```
Comment (DmytroSvirsa) [inline app/main.py:21]: maybe Deck(start[1], row) should be Deck(row, start[1])? it passes tests, but if you print the field, you will see, it's wrong

**Review by DmytroSvirsa — CHANGES_REQUESTED** [review]

**Review by TPolina — APPROVED** [review]

**Pull request closed without merging**